tests/kani/Intrinsics/discriminant_value.rs

```diff
@@ -0,0 +1,50 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+// Check that `discriminant_value` returns the expected results
+// for different cases
+#![feature(core_intrinsics)]
+use std::intrinsics::discriminant_value;
+
+// A standard enum with variants containing fields
+enum MyError {
+    Error1(i32),
+    Error2(&'static str),
+    Error3 { description: String, code: u32 },
+}
+
+// An enum that assigns constant values to some variants
+enum Constants {
+    A = 2,
+    B = 5,
+    C,
+}
+
+// An enum that assigns constant values (one of them negative) to all variants
+enum Ordering {
+    Less = -1,
+    Equal = 0,
+    Greater = 1,
+}
+
+#[kani::proof]
+fn main() {
```

Comment:
Can you please split this into 3 different harnesses? I think it will make this much easier to read.

Reply:
Thanks for the suggestion, I didn't think about that! It's now split into 4 harnesses, can you take another look?